mono/metadata/object.c

```diff
@@ -54,6 +54,13 @@
 #include "cominterop.h"
 #include <mono/utils/w32api.h>
 #include <mono/utils/unlocked.h>
+#include "external-only.h"
+#include "monitor.h"
+
+// Exports do not work portably/reliably if compiled separately.
+// They must be in .o/.obj files otherwise referenced.
+// .def files work better in this regard.
```

Comment:
Maybe it doesn't hurt to be a bit more verbose here.  Something like:

```
// If an object file full of exported functions that are not called anywhere is included in
// an intermediate static library, and the library is in turn linked into another binary or library, the
// external only object fill will be dropped.  As a workaround, we include "external-only.c" here.  Another
// workaround on Windows would be to use .def files to specify exports.  On Unix, we would need to link
// external-only.o directly into the final binary or shared library.
```

Reply:
Unix has .def files or equivalent also.

Reply:
And listing .o files on link command line works on Windows. I'll wordsmith a bit.